urwid/widget/wimp.py

```diff
@@ -29,34 +29,46 @@
 from urwid.util import is_mouse_press
 
 from .columns import Columns
-from .constants import Sizing
+from .constants import Align, Sizing, WrapMode
 from .text import Text
 from .widget import WidgetWrap
 
 if typing.TYPE_CHECKING:
-    from collections.abc import MutableSequence
+    from collections.abc import Callable, MutableSequence
 
-    from typing_extensions import Literal
+    from typing_extensions import Literal, Self
 
     from urwid.canvas import TextCanvas
 
+    _T = typing.TypeVar("_T")
+
 
 class SelectableIcon(Text):
     ignore_focus = False
     _selectable = True
 
-    def __init__(self, text, cursor_position=0):
+    def __init__(
+        self,
+        text,
+        cursor_position: int = 0,
+        align: Literal["left", "center", "right"] | Align = Align.LEFT,
+        wrap: Literal["space", "any", "clip", "ellipsis"] | WrapMode = WrapMode.SPACE,
+    ) -> None:
         """
         :param text: markup for this widget; see :class:`Text` for
                      description of text markup
         :param cursor_position: position the cursor will appear in the
                                 text when this widget is in focus
+        :param align: typically ``'left'``, ``'center'`` or ``'right'``
+        :type align: text alignment mode
+        :param wrap: typically ``'space'``, ``'any'``, ``'clip'`` or ``'ellipsis'``
+        :type wrap: text wrapping mode
```

Comment:
don't need to say "typically" here if there's no way to use something other than StandardTextLayout

Reply:
exposed possibility to override layout. Now fully flexible.